src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1625,7 +1640,16 @@ private static bool QueueUserWorkItemHelper(WaitCallback callBack, Object state,
                 try { }
                 finally
                 {
-                    QueueUserWorkItemCallback tpcallBack = new QueueUserWorkItemCallback(callBack, state, compressStack, ref stackMark);
+                    ExecutionContext context = compressStack && !ExecutionContext.IsFlowSuppressed() ?
+                                                ExecutionContext.Capture(ref stackMark, ExecutionContext.CaptureOptions.IgnoreSyncCtx | ExecutionContext.CaptureOptions.OptimizeDefaultCase) :
+                                                ExecutionContext.Default;
+
+                    QueueUserWorkItemCallback tpcallBack = (context == ExecutionContext.Default) ?
+                                                (state == null ? 
+                                                    new QueueUserWorkItemCallback(callBack) : // default or null context, no state 
+                                                    new QueueUserWorkItemCallbackWithState(callBack, state)) : // default or null context
```

Comment:
TCS created; operation put on a libuv thread; When the operation completes the libuv thread calls the call back; to not block the libuv thread; the TCS SetResult is queued to the threadpool; and the 41MB of allocations is from the `ThreadPool.QueueUserWorkItem(_runAction, action);` but its not going to be capturing a very interesting call stack:

![call-stack](http://aoa.blob.core.windows.net/aspnet/ec.png)

Since the flow can`t be suppressed, would it be better to just embrace it and capture the context at the time of creating the tcs? e.g.
```csharp
tcs = new TaskCompletionSource<object>(ExecutionContext.Capture());`
```
then `Run` it later prior to queuing to threadpool
```csharp
ExecutionContext.Run((ExecutionContext)tcs.Task.AsyncState, 
    o =>
        ThreadPool.QueueUserWorkItem(
            (state) => 
                ((TaskCompletionSource<object>)state).TrySetResult(null), 
                (TaskCompletionSource<object>)o), 
    tcs);
```
Seems a bit strange though?